evaluations/subcoin_2_piewol.md

```diff
@@ -0,0 +1,191 @@
+# Evaluation
+
+- **Status:** in progress
+- **Application Document:** https://github.com/w3f/Grants-Program/blob/master/applications/Subcoin.md
+- **Milestone:** 2
+
+| Number | Deliverable    | Accepted | Link | Notes |
+| -- | --  | ---    | --- | --- |
+| 0a.    | License                 | <ul><li>[x] </li></ul> | https://github.com/subcoin-project/subcoin/blob/subcoin-milestone-2/LICENSE |  --  |
+| 0b.    | Documentation           | <ul><li>[ ] </li></ul> | Each module has its own docs. The rendered inline rustdoc is deployed at https://subcoin-project.github.io/subcoin/.  | some elements e.g. structs are still without inline doc |
+| 0c.    | Testing and Testing Guide | <ul><li>[x] </li></ul> |  https://github.com/subcoin-project/subcoin/tree/subcoin-milestone-2?tab=readme-ov-file#run-tests |  see general notes  |
+| 0d.    | Docker                  | <ul><li>[ ] </li></ul> | https://github.com/subcoin-project/subcoin/blob/subcoin-milestone-2/Dockerfile The docker image is available at https://github.com/subcoin-project/subcoin/pkgs/container/subcoin/249545041?tag=v0.2.0  |  builds. see notes. |
+| 1.     | Block Verification	       | <ul><li>[ ] </li></ul> | 	https://github.com/subcoin-project/subcoin/blob/subcoin-milestone-2/crates/sc-consensus-nakamoto/src/verification.rs |  how comes that there are no tests? Is this directly from ``rust-bitcoin``? |
+| 2.     | `bitcoin-network`       | <ul><li>[x] </li></ul> | https://github.com/subcoin-project/subcoin/blob/subcoin-milestone-2/crates/subcoin-network/src/lib.rs | works, gets peers and imports blocks  |
+
+# General Notes
+
+
+In ``pallet-bitcoin`` you use the error message "txid must be encoded correctly; qed" both for encoding and decoding operations. I think it would be great it these error messages were unique for both operations. Furthermore it looks like the result type in the trait is used for a reason as the writer can emit errors. Whats the reason you put an `.expect()` here allowing for a panic? Basically, why are you sure that it will never happen? 
+
+Are you planning to introduce more tests to vital modules? ``pallet-bitcoin`` currently only has one test to check on encoding. Unit tests should cover the logic you have implemented.
+
+## Documentation
+
+Some elements e.g. structs are still without inline doc. It would be great if this could be completed.
```

Comment:
Just as an example: 
https://subcoin-project.github.io/subcoin/sc_consensus_nakamoto/struct.ImportBlocks.html
https://subcoin-project.github.io/subcoin/pallet_bitcoin/fn.coin_storage_prefix.html

Also private items should be documented. It's not only relevant that people can use any code produced as a library, only understanding public api's. Instead we want to support software that is easy to maintain and understand in case somebody else picks it up for further development.

Reply:
I fully agree with you. Though, how self-explanatory some code really is, often is in the eye of the beholder. I just wanted to inform you about one of our reasons to mandate well documented code. 

Thanks for checking the documentation and extending it.